app/controllers/katello/concerns/hosts_controller_extensions.rb

```diff
@@ -23,6 +26,35 @@ def puppet_environment_for_content_view
           cvpe = Katello::ContentViewPuppetEnvironment.where(:environment_id => environment, :content_view_version_id => version).first
           render :json => cvpe.nil? ? nil : {:name => cvpe.puppet_environment.name, :id => cvpe.puppet_environment.id}
         end
+
+        def content_hosts
+          respond_to do |format|
+            format.csv do
+              @hosts = resource_base_with_search.where(organization_id: params[:organization_id])
+                         .preload(:subscription_status_object, :content_facet, :subscription_facet)
+              csv_response(@hosts,
```

Comment:
Why not use a view layer (i.e. hosts.csv.erb) ? More complex model-formatting logic could go in a decorator or presenter. It seems like bad practice to bloat the controller for csv functionality.